src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -127,9 +134,18 @@ plugin = GHC.defaultPlugin {
           let warning = mkWarning (mkSrcSpan srcLoc srcLoc) msg
           liftIO $ printWarning dynFlags warning
           pure gblEnv
-        else typecheckHook opts summary gblEnv 
-
-
+        else do
+          newGblEnv <- typecheckHook opts summary gblEnv
+          case newGblEnv of
+            Left KillPluginWithSuccess -> pure gblEnv
+            Right (newGblEnv', errorFilters) ->
```

Comment:
The convention in Haskell is to use `Left` to report errors. 
Maybe `LiquidCheckException` could have another case to report filters that haven't been matched.